packages/vscode-extension/src/debug/progressHelper.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IProgressHandler } from "@microsoft/teamsfx-api";
+import { ProgressHandler } from "../progressHandler";
+
+export class DebugProgressHandler extends ProgressHandler implements IProgressHandler {
+  private stepNum: number;
+  constructor(title: string, totalSteps: number) {
+    super(title, totalSteps);
+    this.stepNum = 0;
+  }
+
+  public async next(detail?: string) {
+    ++this.stepNum;
+    super.next(detail);
+  }
+
+  public getStepNum(): number {
+    return this.stepNum;
+  }
+}
+
+export class ProgressBarGroup {
+  private subProgressBars: DebugProgressHandler[] = [];
```

Comment:
Multiple bars may confuse the order, e.g., 3/4 is stilling running but 4/4 is done.

Use one progress bar and always show the running one, e.g.,
<img width="396" alt="image" src="https://user-images.githubusercontent.com/7642967/153536141-d9414419-ba4b-4c80-8911-acf851bd0f38.png">

It's possible that some short operation just show a short bar.